Python/opcode_metadata.h

```diff
@@ -3,15 +3,19 @@
 // Do not edit!
 enum Direction { DIR_NONE, DIR_READ, DIR_WRITE };
 enum InstructionFormat { INSTR_FMT_IB, INSTR_FMT_IBC, INSTR_FMT_IBC0, INSTR_FMT_IBC000, INSTR_FMT_IBIB, INSTR_FMT_IX, INSTR_FMT_IXC, INSTR_FMT_IXC000 };
-static const struct {
+struct opcode_metadata {
     short n_popped;
     short n_pushed;
     enum Direction dir_op1;
     enum Direction dir_op2;
     enum Direction dir_op3;
     bool valid_entry;
     enum InstructionFormat instr_format;
-} _PyOpcode_opcode_metadata[256] = {
+};
+
+static struct opcode_metadata
+_PyOpcode_opcode_metadata(int opcode, int oparg) {
+    struct opcode_metadata metadata[256] = {
```

Comment:
While right now this data isn't used except in that check, the idea is that it would be used (aren't you planning to use it in the optimizer?), and I think the way this works is that each call re-creates the entire array on each call. That would seem inefficient in the extreme.

A more efficient way would be to keep most of the data static, and have a switch on opcode in the generated function. E.g. a function that returns `n_popped`:
```cc
switch (opcode) {
    case NOP: return 0;
    case RESUME: return 0;
    ...
    case LIST_APPEND: return (oparg-1)+2;
    ...
}
```
It would seem important that if all you ask for is static data (e.g. `instr_format`), you don't have to call anything but you can just keep indexing an array.